src/DataMigration/DataMigration/Cmdlets/NewAzureActiveDirectoryAppCmdlet.cs

```diff
@@ -0,0 +1,58 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using System.Security;
+using Microsoft.Azure.Commands.DataMigration.Models;
+
+namespace Microsoft.Azure.Commands.DataMigration.Cmdlets
+{
+    /// <summary>
+    /// Class that creates a new instance of AzureActiveDirectoryApp.
+    /// </summary>
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DataMigrationAzureActiveDirectoryApp", SupportsShouldProcess = true), OutputType(typeof(PSAzureActiveDirectoryApp))]
```

Comment:
@sichoudh what is the expected usage of this cmdlet? Where would the `PSAzureActiveDirectory` object be used once it's created from this cmdlet?

Reply:
This is input needed for MI online Sync migration